components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt

```diff
@@ -466,12 +477,20 @@ open class Nimbus(
 
     override fun setExperimentsLocally(@RawRes file: Int) {
         dbScope.launch {
-            withCatchAll {
-                context.resources.openRawResource(file).use {
-                    it.bufferedReader().readText()
+            setExperimentsFrom(file)
+        }
+    }
+
+    override fun tryLoadAndApplyFirstRunExperiments(@RawRes file: Int, timeout: Long) {
+        runBlocking {
+            try {
+                withTimeout(timeout) {
+                    setExperimentsFrom(file)
+                    applyPendingExperimentsOnThisThread()
```

Comment:
To avoid this consuming resource on the main thread, we could lunch on `dbScope` and join on it until it finish and cancel this job if we timeout, something like


```kotlin
 runBlocking {
        try {
            withTimeout(500) {
                job.join()
            }
        } catch (ex: TimeoutCancellationException) {
            job.cancel()
        }
    }
```

Reply:
woah beat me to it by a few seconds 😬